internal/util/cmdutil/cmdutil.go

```diff
@@ -178,6 +179,44 @@ func CheckDirectoryNotPresent(outDir string) error {
 	return nil
 }
 
+// SuggestFunctions looks for functions from kpt curated catalog list as well as the Porch
+// orchestrator to suggest functions.
+func SuggestFunctions(cmd *cobra.Command) []string {
+	var functions []string
+	porchClient := porchutil.NewPorchFunctionClient()
+	if porchClient.IsPorchAvailable {
+		porchFunctions, err := porchClient.ListFunctions(cmd)
+		if err == nil {
+			functions = append(functions, porchFunctions...)
+		}
+	}
+	// TODO(yuwenma): We should filter these catalog.kpt.fn functions by its annotation as well! This requires changing
+	// the kpt-functions-catalog release workflow to include new annotations.
+	functions = append(functions, FetchFunctionImages()...)
+	var dedup []string
+	fnMap := map[string]bool{}
+	for _, function := range functions {
+		if _, ok := fnMap[function]; !ok {
+			dedup = append(dedup, function)
+			fnMap[function] = true
+		}
+	}
+	return dedup
```

Comment:
IMO, we should fire off both lookups in parallel (porch and functionImages) so that autocomplete UX feels instantaneous (more important from `porch disconnected` mode)

Reply:
I'll do that in the follow up PR since I also plan to make some major changes to the `fetchFunctions` (including OCI annotations and filter functions by annotation)